cnf-tests/pod-utils/cyclictest-runner/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"flag"
+	"k8s.io/klog"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/pod-utils/pkg/node"
+)
+
+const cyclictestBinary = "/usr/bin/cyclictest"
+
+func main() {
+	klog.InitFlags(nil)
+
+	rtPriority := flag.String("rt-priority", "1", "specify the SCHED_FIFO priority (1-99)")
+	duration := flag.Duration("duration", 15 * time.Second, "specify a length for the test run. Append 'm', 'h', or 'd' to specify minutes, hours or days.")
+	threads := flag.String("threads", "", "number of threads: MAX, threads = max_cpus. without -t default = 1")
+	histogram := flag.String("histogram", "30", "dump a latency histogram to stdout after the run US is the max latency time to be be tracked in microseconds")
+	interval := flag.Int("interval", 1000, "base interval of thread in us default=1000")
+	cyclictestStartDelay := flag.Duration("cyclictest-start-delay", 0, "delay in second before running the cyclictest binary")
+
+	flag.Parse()
+
+	selfCPUs, err := node.GetSelfCPUs()
+	if err != nil {
+		klog.Fatalf("failed to get self allowed CPUs: %v", err)
+	}
+
+	if len(*threads) == 0 {
+		*threads = strconv.Itoa(selfCPUs.Size())
```

Comment:
Isn't this different from what the comment says? If we don't pass `-t` then it will be maxCpus I guess?

Reply:
this is how the help from the ```cyclictest``` looks like:
```
-t       --threads         one thread per available processor
-t [NUM] --threads=NUM     number of threads:
                           without NUM, threads = max_cpus
                           without -t default = 1
```
The description of the test itself is a bit unclear I think but this is how under it:
if the ```--thread``` flag is provided without any number then we should run the test on all threads that ```cyclictest``` allowed to run on (i.e. max_cpus) 
without ```--thread``` test should run only on 1 thread.
Now for the wrapper we do provide the ```--thread``` flag without any number, so we want all CPUs (i.e max_cpus).
In addition to that, I guess testing only a single thread on a cluster with quite some CPUs is quite not productive am I right?
now regarding if the flag should be a string or int, I can convert it to an int, if you think it's more clear and use other value than ```""```

Reply:
The description is clear, but with go flags we don't have a way to express that, since I think you can't pass a flag without arguments (if you use `flags`, probably other libraries will allow that).

In the current implementation you always pass `-t` to the binary, so the code must be changed a bit to accomodate that or the comment of the threads flag must be adjusted.

Reply:
OK I think I understand now, so let's do the following:
In case no ```--thread``` flag is provided at all, we'll go with the default which is ```1 thread```
and in order to mimic the max_cpus the user shall provide ```0``` for all, is that works for you?

Reply:
Sounds good.
-1 for max_cpus sounds also like a reasonable option

Reply:
Done